_docs/xcm/integration-overview.md

```diff
@@ -0,0 +1,131 @@
+1. Who are the target audience, or beneficial end users of this product?
+    
+    For example, the audience of Shiden <> Turing integration is all users on Shiden Network.
+2. What would be our product offering?
+    
+    Taking Dapp staking as an example, the feature would automate two operations, claim and deposit.
+
+    Another example would be DEX liquidity pool reward compound, which will include three operations, claim, swap, and deposit.
+3. What are the user defined input?
+
+    For auto-compound, normally the UI will need to ask users to provide some preferences, such as,
+    
+    Frequency: how often to compound
+    
+    Amount: percentage of claimed reward to compound(No need since the current re-deposit ratio is always 100%)
+
+### 1.2 Picture the flow
+After going through design decision calls, we will draw a high-level flow chart to picture the integration flow. For example, below is a diagram of a two-way XCM product integration.
+![Two-way XCM product integration](../../../assets/img/integration-overview/two-way-xcm.png)
+
+
+### 1.3 UI/UX Design
+We will also provide partners with product mockups as recommendation, and to mimic the end user experience as a guideline. For example, below are our mockups for Mangata LP reward auto-compound.
+
+1. After user turns on compound checkbox, navigate to Step 1, a one-time setup for user’s wallet, and ask to sign.
+![Mangata integration Step 1](../../../assets/img/integration-overview/mangata-step-1.jpg)
+
+1. After signing, move on to Step 2 in order to set up automation on Turing Network.
+![Mangata integration Step 2](../../../assets/img/integration-overview/mangata-step-2.jpg)
+
+## 2. Define XCM Message
+We will go over the details of the XCM message design in the XCM Message document, but here’s a template that our automation uses to send the task function call over for the target chain to execute.
+```rust
+let target_xcm =
+Xcm(vec![
+        ReserveAssetDeposited::<()>(local_asset.clone().into()),
+        BuyExecution::<()> { fees: local_asset, weight_limit: Limited(xcm_weight) },
+        DescendOrigin::<()>(descend_location),
+        Transact::<()> {
+            origin_type: OriginKind::SovereignAccount,
+            require_weight_at_most: transact_encoded_call_weight,
+            call: transact_encoded_call.into(),
+        },
+        RefundSurplus::<()>,
+        DepositAsset::<()> {
+            assets: Wild(All),
+            max_assets: 1,
+            beneficiary: MultiLocation {
+                parents: 1,
+                interior: X1(Parachain(T::SelfParaId::get().into())),
+            },
+        },
+    ]);
+```
+
+## 3. Define the fees
+Fee is a big part of the product development, because there are several fees involved in the flow, if not implemented 100% accurately, small errors can accrue into bigger problems, such as insufficient balance for automation fees, or under-utilized residual balances.
```

Comment:
```suggestion
Fees are a big part of the product development, because there are several fees involved in the flow. If not implemented 100% accurately, small errors can accrue into bigger problems, such as insufficient balance for automation fees, or under-utilized residual balances.
```